From f56c42a5057878a196c9d3799a65eb6913060a88 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Wed, 28 Jul 2021 14:28:02 +0300
Subject: [PATCH 24/49] Revert "sram: Add LLCE sram awareness"

commit f56c42a5057878a196c9d3799a65eb6913060a88 from
https://source.codeaurora.org/external/autobsps32/linux

This reverts commit 0fd89599ec5f7e1db229a639985f6468bae647ee.

Upstream-Status: Pending

Issue: ALB-6953
Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/misc/sram.c | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/drivers/misc/sram.c b/drivers/misc/sram.c
index 0f5d3d683325..70a70ae327c1 100644
--- a/drivers/misc/sram.c
+++ b/drivers/misc/sram.c
@@ -3,7 +3,7 @@
  * Generic on-chip SRAM allocation driver
  *
  * Copyright (C) 2012 Philipp Zabel, Pengutronix
- * Copyright 2020-2021 NXP
+ * Copyright 2020 NXP
  */
 
 #include <linux/clk.h>
@@ -318,7 +318,7 @@ static int sram_reserve_regions(struct sram_dev *sram, struct resource *res)
 	return ret;
 }
 
-static int atmel_securam_wait(struct platform_device *pdev)
+static int atmel_securam_wait(void)
 {
 	struct regmap *regmap;
 	u32 val;
@@ -344,7 +344,6 @@ static int llce_init_sram(struct device *dev)
 static const struct of_device_id sram_dt_ids[] = {
 	{ .compatible = "mmio-sram" },
 	{ .compatible = "atmel,sama5d2-securam", .data = atmel_securam_wait },
-	{ .compatible = "nxp,s32g-llce-sram", .data = llce_init_sram },
 	{}
 };
 
@@ -352,7 +351,7 @@ static int sram_probe(struct platform_device *pdev)
 {
 	struct sram_dev *sram;
 	int ret;
-	int (*init_func)(struct device *dev);
+	int (*init_func)(void);
 
 	sram = devm_kzalloc(&pdev->dev, sizeof(*sram), GFP_KERNEL);
 	if (!sram)
@@ -389,7 +388,7 @@ static int sram_probe(struct platform_device *pdev)
 
 	init_func = of_device_get_match_data(&pdev->dev);
 	if (init_func) {
-		ret = init_func(&pdev->dev);
+		ret = init_func();
 		if (ret)
 			goto err_free_partitions;
 	}
-- 
2.17.1

